metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py

```diff
@@ -69,6 +73,13 @@ class BaseSnowflakeConfig(BaseTimeWindowConfig):
 
     scheme: str = "snowflake"
     username: Optional[str] = None
+    use_certificate: Optional[bool] = False
```

Comment:
Should we extract all these `oauth` settings in their own object? Currently it is hard to know what is required. 3 of them are not optional but this makes it hard to see whether they are optional or not.

Reply:
done